src/binary/cursor.rs

```diff
@@ -1063,11 +1107,13 @@ where
     }
 
     fn skip_current_value(&mut self) -> IonResult<()> {
-        if self.cursor.index_at_depth == 0 {
-            Ok(())
-        } else {
+        if self.cursor.index_at_depth > 0
+            && self.cursor.bytes_read < self.cursor.value.value_end_exclusive()
```

Comment:
I think this check is superfluous? `skip_bytes` will early return if `bytes_to_skip` is `0`.

Reply:
We dove on this offline- this check was due to recursive use of `next()` to handle NOP sleds. The method has been clarified.